### PR TITLE
Iptables speedup

### PR DESCRIPTION
**Pull request by DaanHoogland**

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@karuturi I don't think it needs CI (yet), same as with #2083

File: systemvm/patches/debian/config/opt/cloud/bin/cs_iptables_save.py [hunk 1]
```diff
@@ -2,6 +2,22 @@
 #
 # -*- coding: utf-8 -*-
 #
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (DaanHoogland) [inline systemvm/patches/debian/config/opt/cloud/bin/cs_iptables_save.py]: this is a big crime of mine. merge must not happen

Comment (DaanHoogland) [inline systemvm/patches/debian/config/opt/cloud/bin/cs_iptables_save.py]: I mailed the author of iptables_convert , Johannes Hubertz <johannes@hubertz.de>, @sl0, and he agreed to publish his code under apache license version2. so I am forgiven.
He has a focus on security and python iptables utilities so I will work with him on the best integration into CloudStack of his code.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by sl0** [comment]
Thank you very much for considering to use my iptables_converter.py in Apache CloudStack. Originally it was licensed GPLv3, but I agree to have a copy here under Apache Foundation License, as that is a Free Software License as well. 
Have fun!
sl0

**Comment by DaanHoogland** [comment]
a lot of errors. need to investigate. see if i screwed up something during rebasing

**Comment by rhtyd** [comment]
needs review on failing tests, I'll review once the issues are fixed. thanks.

**Comment by rhtyd** [comment]
@DaanHoogland is this still valid, should we consider?

**Comment by rhtyd** [comment]
@DaanHoogland ping

**Comment by DaanHoogland** [comment]
I have no idea nor time to look at those errors. we'll have to rebase and target this for master

File: systemvm/patches/debian/config/opt/cloud/bin/cs_iptables_save.py (hunk 1 above)
Comment (rhtyd) [inline systemvm/patches/debian/config/opt/cloud/bin/cs_iptables_save.py]: @DaanHoogland any update on this?

**Comment by rhtyd** [comment]
Thanks @sl0, can you re-publish the source file under Apache License 2.0 in your repo that we can use, or send to us as a pull request?

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by sl0** [comment]
Hello Rohit,

On 08/19/2017 11:32 PM, Rohit Yadav wrote:
> Thanks @sl0 <https://github.com/sl0>, can you re-publish the source file
> under Apache License 2.0 in your repo that we can use, or send to us as
> a pull request?

You just not want to use the whole package, just part of it. And Daan
asked me th have it with Apache License. I agreed and he has changed in
your repo and I agreed again.

Is this not enough for you? I've put GPL on my sources to prevent them
from been put into a commerial cloesed source project legally. The
inheritance priciple of GPL is very important for me.

FYI: As I worked a long time on it and then continued to have it in
Debian, I'm afraid of changing the license to any other. And today it is
contained in the Debian stretch release, if you like, take it as a
debian package as an external component?

I totally agree to have parts of my iptables-converter within Apache
Cloudstack relicensed to Apache License, but I would be very unhappy to
change GPL on my sources at github, because I'm afraid I have to leave
the Debian Friends...

Thanks for reading, kind regards from Cologne, Germany.
Thank you in advance for any response.

Johannes
(sl0)

**Comment by rhtyd** [comment]
Hello Johannes ( @sl0 ),
Thank you for replying. Sorry for the confusion, I'm not requesting you to move towards a new license but dual-license your work, i.e. both your GPL and Apache license. It would be best if you do that in your repository so we can point to your work in case of any issues. One way to do that is to include the Apache License in your repo's root folder and include the Apache license on all the file headers like it's been done in this PR. You may, for example, look at https://github.com/facebook/rocksdb/tree/master and others who have adopted a dual-license approach. Thanks a lot for the conversation and your swift reply.

Regards, Rohit (@rhtyd).

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by agx** [comment]
@DaanHoogland  Wouldn't it be nicer if cloudstack would pull this in via either pypi or as a git submodule? This way diverging of the codebases would be less likely and bugs would be fixed in one place?

**Comment by kiwiflyer** [comment]
@DaanHoogland What can we do to help with this one? It looks really worth while. 

@dmabry

**Comment by agx** [comment]
@kiwiflyer if you're fine with adding iptables-opti to your python requirements.txt I can work with @sl0 to turn this into a pypi'able module (if that's o.k. for him)

**Comment by DaanHoogland** [comment]
@kiwiflyer this has gone completely from my radar, sorry. I would like to see it in, but have no time to look at it for a while. It is not my code so I'd have to study on it. @sl0's module as a pypi thingy makes sense but to us it will require a refactor of the code, removing the code copy and adding the install in the systemvm build.

**Comment by agx** [comment]
@DaanHoogland from what I can see to use iptables-conv verbatim we'd mostly need to 

* not sys.exit(1) from the module (done)
* move to using logger instead of print
* be able to write a file instead of stdout

Did I miss s.th.?

**Comment by sl0** [comment]
Hellllo @DaanHoogland and @rhtyd ,
a new version 0.910 of iptables-converter is available. Yesterday it was released at pypi: https://pypi.python.org/pypi/iptables-converter/0.9.10 Now it ships as a python-module with entry-points. Some documentation is prepared: https://conv.readthedocs.io/en/latest/
Some more tests cover 97% now and flake8 silence are added as well. python 2.7, 3.5 and 3.6 work with tox as testrunner usind py.test . Anything missing? Thank you very much for your answer in advance
Kind regards from Cologne
Johannes
(sl0)

**Comment by rhtyd** [comment]
Thanks @sl0 I've pulled the updated release, we'll work on refactoring it if needed.
@blueorangutan package

**Comment by rhtyd** [comment]
Related for reference: https://github.com/apache/cloudstack/pull/1400/files

**Comment by rhtyd** [comment]
Due to lack of activity, I've removed the 4.11 milestone. We can re-engage and work on this for a future minor 4.11.x release.

**Comment by borisstoyanov** [comment]
@DaanHoogland do you think this would make it in 4.11.1? if not we can drop it from the list perhaps?

**Pull request closed without merging**